cmd/sentinel/cmd/sentinel.go

```diff
@@ -689,12 +690,17 @@ func (s *Sentinel) validStandbysByStatus(cd *cluster.ClusterData) (map[string]*c
 	return goodStandbys, failedStandbys, convergingStandbys
 }
 
-// dbSlice implements sort interface to sort by XLogPos
-type dbSlice []*cluster.DB
-
-func (p dbSlice) Len() int           { return len(p) }
-func (p dbSlice) Less(i, j int) bool { return p[i].Status.XLogPos < p[j].Status.XLogPos }
-func (p dbSlice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+// sort dbs by XLogPos and keeper's priority
+func sortDBs(cd *cluster.ClusterData, dbs []*cluster.DB) {
+	sort.Slice(dbs, func(i, j int) bool {
+		if dbs[i].Status.XLogPos != dbs[j].Status.XLogPos {
+			return dbs[i].Status.XLogPos < dbs[j].Status.XLogPos
+		}
+		pi := cd.Keepers[dbs[i].Spec.KeeperUID].Status.Priority
+		pj := cd.Keepers[dbs[i].Spec.KeeperUID].Status.Priority
+		return pi < pj
```

Comment:
In Line 700,
 ```
pj := cd.Keepers[dbs[i].Spec.KeeperUID].Status.Priority
```
replace `dbs[i] to dbs[j]`

Reply:
Ups, good catch, thanks.